packages/blocks/src/page-block/default/utils.ts

```diff
@@ -136,13 +136,23 @@ function binarySearchBlockEditingState(
     containerLeft = firstBlock.blockRect.left;
   }
 
+  let inside = false;
   while (start <= end) {
     const mid = start + Math.floor((end - start) / 2);
     const { block, blockRect, detectRect, hoverDom } = getBlockAndRect(
       blocks,
       mid
     );
 
+    // if the detectRect is not in the view port, it's definitely not the block we want
+    if (detectRect.top > window.innerHeight) {
+      end = mid - 1;
```

Comment:
I think we can add a comment here to use the `container.bottom` and `container.top` to determine if further optimization is needed.

Reply:
Does the `container` here mean `detectRect`?

Reply:
`window.innerHeight` and `0`

Reply:
`clientY` and `detectRect.top` are both relative to viewport, I think `window.innerHeight` and `0` should work here? But I'm not sure (

Reply:
Sometimes `.affine-default-viewport` does not fill the entire window.

Reply:
I think viewport here means  **device viewport** used by `clientY` and `detectRect.top` unless it's embedded in an `iframe`

This contributes to the clarity and ease of use of the current method

Reply:
It should be the viewport of the scrolling container.
<img width="1072" alt="Screenshot 2023-03-04 at 11 24 39 AM" src="https://user-images.githubusercontent.com/27926/222873465-cc7f1cb2-821c-44f2-a623-cd5d5ed21872.png">

Reply:
Got it, I only considered the consistency of two coordinates